### PR TITLE
Fix #1292: nocolor and verbose can work in config files

### LINKED ISSUE #1292: verbose and nocolor options are ignored from config files
## Expected Behaviour

The [default configuration file](https://docs.sqlfluff.com/en/stable/configuration.html#default-configuration) has the following entries:

```
verbose = 0
nocolor = False
```

This suggests these options can be configured in a config file but this does not appear to be the case.

## Observed Behaviour

Changing those defaults does not change the verbosity nor whether color printing is used.

## Steps to Reproduce

- Create a .sqlfluff files with the following set:

```
[sqlfluff]
verbose = 1
nocolor = True
```

- Lint files.
- The verbosity is not used and the default color options are used.

## Version
Include the output of `sqlfluff --version` along with your Python version

Latest (0.6.3) and Python 3.8

## Configuration

```
[sqlfluff]
verbose = 1
nocolor = True
```

## More information

If you add the following to `def common_options` for both these config in [`src/sqlfull/cli/commands.py`](https://github.com/sqlfluff/sqlfluff/blob/00fca3a3af9d48902e35a6de8fd60aff32de2035/src/sqlfluff/cli/commands.py#L104):
```
        default=None,
```

Then it looks like these configs can be overridden by a config file in the current directory. Without this, it looks like the command line options are used (even if not set).

Is there a reason these are not set with this `default=None` option?

However even with this change, it looks like it's only honoured for the config file in the current directory that sqlfluff is run from. And if you specify a path (e.g. `sqlfluff lint /some/other/path`) then the configs are not used for these two settings (even though other settings are used for linting/parsing/fixing). I'm not sure if this is due in part due to the issue discussed in #556 ?

I'm not sure it makes sense to override some of these options anyway - e.g. does it make sense to have different verbosity or nocolor settings for different paths or should it be a global setting?

Maybe, rather than looking to fix this, these two options should these just be made command line options and removed from the example default config file so people don't presume they can override this (when currently they can't)?

### PR DESCRIPTION
**Pull request by cympfh**
### Brief summary of the change made

- Fix #1292

### Are there any other side effects of this change that we should be aware of?

The cause of this bug is that some options in Click (`--nocolor` and `--verbose`) don't have default values when not specified as CLI options. In original code, only `None` must be default value. This change should define default values for each options other than `None`.

### Pull Request checklist
- [x] Please confirm you have completed any of the necessary steps below.

- Included test cases to demonstrate any code changes, which may be one or more of the following:
  - `.yml` rule test cases in `test/fixtures/rules/std_rule_cases`.
  - `.sql`/`.yml` parser test cases in `test/fixtures/dialects` (note YML files can be auto generated with `tox -e generate-fixture-yml`).
  - Full autofix test cases in `test/fixtures/linter/autofix`.
  - Other.
- Added appropriate documentation for the change.
- Created GitHub issues for any relevant followup/future enhancements if appropriate.

### CODE REVIEW

File: src/sqlfluff/cli/commands.py
Comment (jpy-git) [inline src/sqlfluff/cli/commands.py:314]: The reason for using `None` as a default value in python is that it is a good way to indicate that a user has not specified an argument versus them explicitly specifying the default value.

If you read your test case, you explicitly pass in `nocolor=False` and receive out `nocolor==True` which is odd behaviour for a function (even if it is doing the correct thing from the CLI perspective):
```python
def test_cli_get_config():
    config = get_config(
        "test/fixtures/config/toml/pyproject.toml",
        True,
        nocolor=False,
        verbose=0,
    )
    assert config.get("nocolor") is True
    assert config.get("verbose") == 2
```

What should happen is if you pass in `nocolor=True` you receive out `True`, if you pass in `nocolor=False` you receive out `False`, and if you pass in `None` (the default) you receive out the value from your config (or the default config if not specified in user config).

File: test/cli/commands_test.py
Comment (jpy-git) [inline test/cli/commands_test.py:935]:
```suggestion
def test_cli_get_config():
    config = get_config(
        "test/fixtures/config/toml/pyproject.toml",
        True,
        nocolor=None,
        verbose=None,
    )
    assert config.get("nocolor") is True
    assert config.get("verbose") == 2
```
See previous review comment for reason

File: src/sqlfluff/cli/commands.py
Comment (jpy-git) [reply src/sqlfluff/cli/commands.py:314]: Can we reset this section to as it was before and instead just simply add `default=None` to the CLI options:
```python
def common_options(f: Callable) -> Callable:
    """Add common options to commands via a decorator.

    These are applied to all of the cli commands.
    """
    f = click.version_option()(f)
    f = click.option(
        "-v",
        "--verbose",
        count=True,
        default=None,
        help=(
            "Verbosity, how detailed should the output be. This is *stackable*, so `-vv`"
            " is more verbose than `-v`. For the most verbose option try `-vvvv` or `-vvvvv`."
        ),
    )(f)
    f = click.option(
        "-n",
        "--nocolor",
        is_flag=True,
        default=None,
        help="No color - if this is set then the output will be without ANSI color codes.",
    )(f)

    return f
```

**Review by jpy-git — COMMENTED** [review]
@cympfh Thanks for spotting the cause of this issue! 😄

Small tweak to make in terms of implementation but otherwise great

**Comment by cympfh** [comment]
Thank you for your comments. Yeah, `default=None` works well!

File: test/cli/commands_test.py
Comment (jpy-git) [inline test/cli/commands_test.py:927]:
```suggestion
def test_cli_get_default_config():
    """`nocolor` and `verbose` values loaded from config if not specified via CLI."""
```
Linting is failing since we need a docstring here.

(for future reference would recommend using the pre-commit hooks to help catch these at commit time: `tox -e pre-commit -- install` 😄)

**Comment by jpy-git** [comment]
@cympfh FYI your PR is correct, the reason it's failing the unit test atm is that we weren't previously allowing `pyproject.toml` files in the `extra_config_path` parameter so it's being ignored. I've create a PR (#2305) to allow this and will unblock this PR once merged in 😄

**Review by jpy-git — APPROVED** [review]
All good, thanks @cympfh 😄 LGTM 🚀 ✨

**Pull request merged**
